x-pack/plugins/security_solution/public/management/pages/event_filters/store/middleware.ts

```diff
@@ -235,6 +235,7 @@ const refreshListDataIfNeeded: MiddlewareActionHandler = async (store, eventFilt
         // Ignore will be fixed with when AsyncResourceState is refactored (#830)
         // @ts-ignore
         type: 'LoadingResourceState',
+        // @ts-ignore
```

Comment:
An issue with `AsyncResourceState` forces to use @ts-ignore in all the places where the LoadingResourceState is used

Reply:
I'm not 100% sure if this is the same problem @paul-tavares was referred to.

Reply:
if is not, then is a new issue related to the asyncresourcestate that I could not resolve with types in this PR.

Reply:
nit/philosophy:

I wouldn't use `@ts-ignore` anymore and would encourage anyone and everyone to not use it anymore. The replacement which is better is `@ts-expect-error`: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#-ts-expect-error-comments

It will give errors once you have fixed the issue which is a better version of `ts-ignore`. We have gone through and replaced all of these once with `ts-expect-error` before but the ts-ignore is showing up again. We might very soon figure out how to ban it and suggest `ts-expect-error` as a linter rule.

As an aside I would also not use `ts-expect-error` either :-) except for my last and only option. If you could cast this as an unknown and then into the expected type that would be better.

The typical mindset I think is best typing to least typing:

Normal typing > as Type > as unknown as Type > ts-expect-error > ts-ignore 

There are a few exceptions to this rule such that `ts-expect-error` is better but really I can't see a legitimate reason to use ts-ignore anymore instead of at least ts-expect-error

Also, although it's a private repo I would expand 830 to be:
https://github.com/elastic/security-team/issues/830

But I would encourage when and where possible to use public repo's for tickets except for road map things to keep the community and users in the loop about public things.

For this I would recommend doing this pattern at least:

```ts
        // @ts-expect-error-next-line will be fixed with when AsyncResourceState is refactored https://github.com/elastic/security-team/issues/830
previousState: getCurrentListPageDataState(state),
```

Reply:
I didn't know `ts-expect-error` existed. Thanks for the tip. I'll try to change it